sktime/forecasting/model_selection/_split.py

```diff
@@ -412,41 +415,211 @@ def __init__(
         self.fh = fh
 
     def split(self, y: ACCEPTED_Y_TYPES) -> SPLIT_GENERATOR_TYPE:
-        """Split `y` into training and test windows.
+        """Get iloc references to train/test slits of `y`.
 
         Parameters
         ----------
-        y : pd.Series or pd.Index
-            Time series to split
+        y : pd.Index or time series in sktime compatible time series format (any)
+            Index of time series to split, or time series to split
+            If time series, considered as index of equivalent pandas type container:
+                pd.DataFrame, pd.Series, pd-multiindex, or pd_multiindex_hier mtype
 
         Yields
         ------
-        train : np.ndarray
-            Training window indices
-        test : np.ndarray
-            Test window indices
+        train : 1D np.ndarray of dtype int
+            Training window indices, iloc references to training indices in y
+        test : 1D np.ndarray of dtype int
+            Test window indices, iloc references to test indices in y
         """
-        y_index = get_index_for_series(y)
-        for train, test in self._split(y_index):
-            yield train[train >= 0], test[test >= 0]
+        y_index = self._coerce_to_index(y)
+
+        if not isinstance(y_index, pd.MultiIndex):
+            for train, test in self._split(y_index):
+                yield train[train >= 0], test[test >= 0]
+
+        if isinstance(y_index, pd.MultiIndex):
```

Comment:
may I ask, why not `elif`?

Reply:
also, why not pack everything after `if` into private methods, say, `_split_flat_index` and `_split_multi_index`? just to improve readability

Reply:
> may I ask, why not elif?

should be a simple else, no? thought, for readability. I added a comment now.

Reply:
> also, why not pack everything after if into private methods, say, `_split_flat_index` and `_split_multi_index`? just to improve readability

Yes, that is the "obvious" best-practice thing to do here.
I am, however, not sure how to structure it, or whether this can be refactored easily with `VectorizedDF`.

My plan was to refactor this to sth more cleaner once it works together with metrics and tuning, but let me know if you have a good idea for function signatures and utilities here.

Reply:
ok, nevermind about `if ... else`. this issue is too small to block anything here.

regarding my second comment. I probably did not state it clear enough. My suggestion is very simple. Everything that goes after `if isinstance(y_index, pd.MultiIndex):` move into a separate method, so that you have something like:
```
if isinstance(y_index, pd.MultiIndex):
    yield self._split_multi_index(y_index)
```
Nothing more complicated than that.

Reply:
> ok, nevermind about if ... else. this issue is too small to block anything here.

Already changed it and added a comment

Reply:
> Everything that goes after `if isinstance(y_index, pd.MultiIndex):` move into a separate method

oh, that makes sense. And then we can refactor the "multiindex" function without affecting `_split`. Good idea, done.